fix(java): correctly inherit properties from parent fields for pom.xml files

DmitriyLewen opened the pull request:
## Description
Maven has 2 "types" of properties:
- from `Properties` element
- from pom fields (e.g. `project.version`, `parent.project.artifactID`, etc.)

Maven priority (check next point if property is empty):
1. field property from pom file
2. field property from parent
3. property from `property` element of pom file (e.g. when `property` map contains explicitly stated property)
4. property from `property` element of parent file (e.g. when `property` map contains explicitly stated property)

### Changes
We need to use properties from `property` element (current pom or parent poms (we merge them when resolving parents)) only if properties from fields are empty.

### Example
See pom files in changed [parent-child-properties](https://github.com/DmitriyLewen/trivy/tree/57bcb28777ecd6d9eab3292ec1db4d12d47739ba/pkg/dependency/parser/java/pom/testdata/parent-child-properties) test.

before:
```json
{
  "Target": "pom.xml",
  "Class": "lang-pkgs",
  "Type": "pom",
  "Packages": [
    {
      "ID": "com.example:child:1.2.3",
      ...
      "Relationship": "root",
      "DependsOn": [
        "org.example:example-dependency:1.2.3"
      ],
      "Layer": {}
    },
    {
      "ID": "org.example:example-api3",
      ...
      "Relationship": "direct",
      "Locations": [
        {
          "StartLine": 30,
          "EndLine": 34
        }
      ]
    },
    {
      "ID": "org.example:example-dependency:1.2.3",
      ...
      "Relationship": "direct",
      "DependsOn": [
        "org.example:example-api:4.0.0"
      ],
      "Locations": [
        {
          "StartLine": 25,
          "EndLine": 29
        }
      ]
    },
    {
      "ID": "org.example:example-api:4.0.0",
      ...
      "Relationship": "indirect",
    }
  ]
}
```
after:
```json
{
  "Target": "pom.xml",
  "Class": "lang-pkgs",
  "Type": "pom",
  "Packages": [
    {
      "ID": "com.example:child:1.2.3",
      ...
      "Relationship": "root",
      "DependsOn": [
        "org.example:example-api3:4.0.3",
        "org.example:example-dependency:1.2.3"
      ],
    },
    {
      "ID": "org.example:example-api3:4.0.3",
      ...
      "Relationship": "direct",
      "Locations": [
        {
          "StartLine": 30,
          "EndLine": 34
        }
      ]
    },
    {
      "ID": "org.example:example-dependency:1.2.3",
      ...
      "Relationship": "direct",
      "DependsOn": [
        "org.example:example-api:4.0.0"
      ],
      "Locations": [
        {
          "StartLine": 25,
          "EndLine": 29
        }
      ]
    },
    {
      "ID": "org.example:example-api:4.0.0",
      ...
      "Relationship": "indirect",
    }
  ]
}
```

mvn dep tree:
```bash
[INFO] com.example:child:jar:1.2.3
[INFO] +- org.example:example-dependency:jar:1.2.3:compile
[INFO] |  \- org.example:example-api:jar:4.0.0:compile
[INFO] \- org.example:example-api3:jar:4.0.3:compile
```


## Related issues
- Close #9067


## Checklist
- [x] I've read the [guidelines for contributing](https://trivy.dev/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://trivy.dev/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

SemProvoost reviewed (approved):
Something we can do to get this one merged? @DmitriyLewen 🎉

SemProvoost commented:
@knqyf263 Any chance you can review this pull request? 😁
Original thread: https://github.com/aquasecurity/trivy/discussions/9039

knqyf263 commented:
We recently launched [our partner program](https://www.aquasec.com/news/trivy-partner-connect-launches-to-expand-open-source-security-scanning-ecosystem/), and since several companies have already joined, we are currently prioritizing their requests. Once those tasks have settled down, we would like to review this PR.

DmitriyLewen commented:
Initially, I made a separate test.
But since there are already too many tests, I found that I could update the current test case and I made minimal changes.

But you're right - now the test case can be simplified.

I removed top-parent and moved some fields to parent.
Now the structure looks like this:
```
parent/pom.xml (inherits top-parent)
├── parent.version: "" ← there is no parent - empty value
├── version: 4.0.3
├── properties:
│   ├── bom.version: 3.0.0
│   └── dep.version: 1.2.3
│
└── child/pom.xml (inherits parent)
    ├── parent.version: 4.0.3  ← references parent
    ├── version: 1.2.3
    ├── properties:
    │   ├── bom.version: 4.0.0
    │   ├── project.parent.version: 1.1.1  ← key point
    │   └── project.version: 2.3.4         ← key point
    │
    └── dependencies:
        ├── example-dependency: ${project.version}
            └── project.version: ${api.version} ←${api.version} (4.0.0) from bom:4.0.0
        └── example-api3: ${project.parent.version}

```
